client_test.go

```diff
@@ -187,12 +305,52 @@ func TestTriggerBatchSuccess(t *testing.T) {
 
 	u, _ := url.Parse(server.URL)
 	client := Client{AppID: "appid", Key: "key", Secret: "secret", Host: u.Host}
-	err := client.TriggerBatch([]Event{
-		{"test_channel", "test", "yolo1", nil},
-		{"test_channel", "test", "yolo2", nil},
+	response, err := client.TriggerBatch([]Event{
+		{Channel: "test_channel", Name: "test", Data: "yolo1"},
+		{Channel: "test_channel", Name: "test", Data: "yolo2"},
+	})
+
+	assert.NoError(t, err)
+	assert.Equal(t, &TriggerChannelsList{}, response)
+}
+
+func TestTriggerBatchInfoSuccess(t *testing.T) {
+	expectedBody := `{"batch":[{"channel":"presence-test_channel","name":"test","data":"yolo1","info":"user_count,subscription_count"},{"channel":"test_channel","name":"test","data":"yolo2","info":"subscription_count"}]}`
+	server := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		res.WriteHeader(200)
+		testJSON := "{\"channels\":{\"presence-test_channel\":{\"subscription_count\":2,\"user_count\":1},\"test_channel\":{\"subscription_count\":3}}}"
+		fmt.Fprintf(res, testJSON)
+		assert.Equal(t, "POST", req.Method)
+
+		actualBody, err := ioutil.ReadAll(req.Body)
+		assert.Equal(t, expectedBody, string(actualBody))
+		assert.Equal(t, "application/json", req.Header["Content-Type"][0])
+		assert.Equal(t, "/apps/appid/batch_events", req.URL.Path)
+		assert.NoError(t, err)
+	}))
+	defer server.Close()
+
+	u, _ := url.Parse(server.URL)
+	client := Client{AppID: "appid", Key: "key", Secret: "secret", Host: u.Host}
+	presenceChannelInfo := "user_count,subscription_count"
+	channelInfo := "subscription_count"
+	channels, err := client.TriggerBatch([]Event{
+		{Channel: "presence-test_channel", Name: "test", Data: "yolo1", Info: &presenceChannelInfo},
+		{Channel: "test_channel", Name: "test", Data: "yolo2", Info: &channelInfo},
 	})
 
 	assert.NoError(t, err)
+
+	presenceExpectedUserCount := 1
+	presenceExpectedSubscriptionCount := 2
+	expectedSubscriptionCount := 3
+	expected := &TriggerChannelsList{
+		Channels: map[string]TriggerChannelListItem{
+			"presence-test_channel": {UserCount: &presenceExpectedUserCount, SubscriptionCount: &presenceExpectedSubscriptionCount},
+			"test_channel":          {SubscriptionCount: &expectedSubscriptionCount},
+		},
+	}
+	assert.Equal(t, expected, channels)
 }
 
```

Comment:
Is there a test for channels with no subscribers? Not sure if i'm missing it but couldn't see one. Other than that looks sensible to me

Reply:
I don't think it is too useful to test this case in the SDK. The response for channels when there are subscribers is:

```json
{
  "channels": {
    "example": {
      "subscription_count": 13
    }
  }
}
```

and when there are no subscribers the response is

```json
{
  "channels": {
    "example": {
      "subscription_count": 0
    }
  }
}
```

so the test would really just be checking that `0` is handled correctly by the Go json library.

I think we _do_ want to add tests to the server to check it returns the response we expect when there are no subscribers.

Reply:
Makes sense to me